### DISCUSSION TITLE
Can't add a first line indent to each bibliography item

### DISCUSSION THREAD

**Discussion by Andrew15-5** [zepinglee/citeproc-lua#60] [Q&A] [answered]
```latex
\documentclass[a4paper]{article}
\usepackage{fontspec}
\usepackage{enumitem}
\setmainfont{Libertinus Serif}
\usepackage[
  style = custom-numeric-unordered-indent,
  % bib-hang = -1em,
  bib-item-sep = 0em,
]{citation-style-language}
\addbibresource{main.bib}
\nocite{*}
\begin{document}
\printbibliography

\begin{enumerate}[wide]
  \item Lorem ipsum dolor sit amet, qui minim labore adipisicing minim sint cillum sint consectetur cupidatat.
  \item Lorem ipsum dolor sit amet, qui minim labore adipisicing minim sint cillum sint consectetur cupidatat.
\end{enumerate}

Lorem ipsum dolor sit amet, officia excepteur ex fugiat reprehenderit enim labore culpa sint ad nisi Lorem pariatur mollit ex esse exercitation amet. Nisi anim cupidatat excepteur officia. Reprehenderit nostrud nostrud ipsum Lorem est aliquip amet voluptate voluptate dolor minim nulla est proident. Nostrud officia pariatur ut officia. Sit irure elit esse ea nulla sunt ex occaecat reprehenderit commodo officia dolor Lorem duis laboris cupidatat officia voluptate. Culpa proident adipisicing id nulla nisi laboris ex in Lorem sunt duis officia eiusmod. Aliqua reprehenderit commodo ex non excepteur duis sunt velit enim. Voluptate laboris sint cupidatat ullamco ut ea consectetur et est culpa et culpa duis.
\end{document}
```

How can I get the same effect of hanging indent 0/width 100% and first line indent to some specific value?

I tried this style in Typst and there you can change the indent with `set par(first-line-indent: 1em)`. I didn't find in the CSL spec if I can add any/specific indent there. And `enumerate`'s indentation, `\indent` or `\hspace` don't work.

Or, as a workaround, I can pass a negative hanging indent, but then I need to somehow move the whole bibliography list to the right. I don't know what are my options.

[lualatex-bibliography.zip](https://github.com/zepinglee/citeproc-lua/files/15041991/lualatex-bibliography.zip)

**Comment by zepinglee** [comment]
I've not implemented the "normal paragraph" style you mentioned.

The CSL has only `hanging-indent` and `second-field-align` to control the printing style of bibliography but it has no access to the specific format like the length of indent.

**Reply by Andrew15-5** [reply]
Ok. Does this mean that currently there is no workaround to achieve the desired output?

**Reply by zepinglee** [reply, marked as answer]
This feature is just added in ff3057d.

You need to remove the `hanging-indent="true"` in `custom-numeric-unordered-indent.csl` and add `bib-par-indent = 1em` in `\cslsetup`. The output of your example is as follows.

<img width="595" alt="Screenshot 2024-04-21 at 00 12 15" src="https://github.com/zepinglee/citeproc-lua/assets/12290822/53abc30c-bd67-4469-a270-eec42c77888b">

**Reply by Andrew15-5** [reply]
Wow, thanks for that! Yep, it works now.

```latex
\setlength\parindent{1em}
\usepackage[
  style = custom-numeric-unordered,
  bib-par-indent = \parindent,
  bib-item-sep = 0em,
]{citation-style-language}
```
